pkg/database/databasescriptview.go

```diff
@@ -0,0 +1,195 @@
+	}
+	sort.Ints(depthsFound)
+	for _, depth := range depthsFound {
+		tableNames := completedTableDepthMap[depth]
+		lineNumbers := []int32{}
+		entityNames := []string{}
+		lineNumberMap := make(map[int32]string)
+		for _, tableName := range tableNames {
+			table := tableMap[tableName]
+			lineNumber := table.GetSourceContext().GetStart().GetLine()
+			lineNumberMap[lineNumber] = tableName
+			lineNumbers = append(lineNumbers, lineNumber)
+		}
+		sort.Slice(lineNumbers, func(i, j int) bool { return lineNumbers[i] < lineNumbers[j] })
+		for _, lineNo := range lineNumbers {
+			entityName := lineNumberMap[lineNo]
+			entityNames = append(entityNames, entityName)
+		}
+		for _, entityName := range entityNames {
+			entityType := tableMap[entityName]
+			if relEntity := entityType.GetRelation(); relEntity != nil {
+				v.writeCreateSQLForATable(entityName, relEntity, visitedAttributes)
+			}
+		}
+	}
+	return v.stringBuilder.String()
+}
+
+func (v *ScriptView) ProcessModSysls(appsOld, appsNew map[string]*sysl.Application,
+	appNames []string, outputDir, dbType string) []ScriptOutput {
+	outputSlice := []ScriptOutput{}
+	for _, appName := range appNames {
+		appOld := appsOld[appName]
+		appNew := appsNew[appName]
+		if appOld != nil && appNew != nil {
+			v.stringBuilder.Reset()
+			typeMapOld := appOld.GetTypes()
+			typeMapNew := appNew.GetTypes()
+			tableDepthMapOld := CreateTableDepthMap(typeMapOld)
+			tableDepthMapNew := CreateTableDepthMap(typeMapNew)
+			tablesWithActions := findAddedDeletedRetainedTables(typeMapOld, typeMapNew, tableDepthMapOld, tableDepthMapNew)
+			outStr := v.processTablesForModifiedApps(tablesWithActions, v.title, appName, dbType)
+			outputFile := filepath.Join(outputDir, appName+SQLExtension)
+			outputStruct := MakeScriptOutput(outputFile, outStr)
+			outputSlice = append(outputSlice, *outputStruct)
+		} else if appNew != nil && appOld == nil {
+			v.stringBuilder.Reset()
+			outStr := v.GenerateDatabaseScriptCreate(appNew.GetTypes(), dbType, appName)
+			outputFile := filepath.Join(outputDir, appName+SQLExtension)
+			outputStruct := MakeScriptOutput(outputFile, outStr)
+			outputSlice = append(outputSlice, *outputStruct)
+		}
+	}
+	return outputSlice
+}
+
+func (v *ScriptView) processTablesForModifiedApps(tableDetails []TableDetails,
+	title, appName, dbType string) string {
+	outputStr := v.generateDatabaseScriptModify(tableDetails, dbType, appName)
+	return outputStr
+}
+
+func findAddedDeletedRetainedTables(
+	tableMapOld map[string]*sysl.Type,
+	tableMapNew map[string]*sysl.Type,
+	tableDepthMapOld, tableDepthMapNew map[int][]string) []TableDetails {
+	tableDepthsListNew := makeSortedListOfTableDepth(tableDepthMapNew)
+	tableWithAction := []TableDetails{}
+	//Add the retained and added tables
+	for _, depth := range tableDepthsListNew {
+		tableNames := tableDepthMapNew[depth]
+		sort.Strings(tableNames)
+		for _, tableName := range tableNames {
+			found := ifTableExisted(tableName, tableMapOld)
```

Comment:
ifTableExisted you probably don't need a method here: just the check the return values of the map membership operation:
if found := tableMapOld[tableName]; found {
tableDetails := MakeTableDetails(tableMapNew[tableName],
					tableMapOld[tableName], Retain, tableName)
				tableWithAction = append(tableWithAction, *tableDetails)
return tableWithAction
}
tableDetails := MakeTableDetails(tableMapNew[tableName],
					nil, Add, tableName)
				tableWithAction = append(tableWithAction, *tableDetails)
return tableWithAction

Reply:
Removed the method